ginga/Control.py

```diff
@@ -850,16 +866,22 @@ def add_image(self, imname, image, chname=None, silent=False):
         if chname is None:
             channel = self.get_channelInfo()
             if channel is None:
-                raise ValueError("Need to provide a channel name to add the image")
+                raise ValueError("Need to provide a channel name to add "
+                                 "the image")
             chname = channel.name
 
+        # Initialize MODIFIED header keyword for image-modified callback.
+        # This is for consistent keyword display in Contents, and used in
+        # 'image-modified' callback.
+        image.metadata['header']['MODIFIED'] = None
```

Comment:
That's probably even more reason to use a "regular" metadata element, since all images have that.  Could you try a test modification of your PR with commit 2931482903b4a61708ddd1928d047b8988b5685a?  This adds the "time_modified" metadata item to `BaseImage` and sets it when the image is modified with `set_data()`.  Will that work fairly cleanly with your PR?